### PR TITLE
Add testing to CI & enable test.sh

### PR DESCRIPTION
**Pull request by garettarrowood**
Tests now run in CI.

Some binstubs are generated.

Invoking `./test.sh` now works, but it does not pass. It fails with a couple dozen rubocop violations.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -24,15 +24,16 @@ app.build(application: application, cluster: cluster, deployerVersion: deployerV
...
   app.dockerStage('Container Build') {
-    sh "docker build -t quay.io/powerhome/${application}:${tag} ."
-    sh "docker push quay.io/powerhome/${application}:${tag}"
+    sh "docker build -t ${appImage} ."
+    sh "docker push ${appImage}"
   }
 
   stage('Test') {
-    // sh "docker run --tty --rm ${appImage.id} bin/rake"
+    sh "docker run --tty --rm ${appImage} bash -lc 'bin/test'"
```
Comment (benlangfeld) [inline Jenkinsfile]: Why bash here?

File: Jenkinsfile [hunk 2]
```diff
@@ -24,15 +24,16 @@ app.build(application: application, cluster: cluster, deployerVersion: deployerV
...
   }
 
   app.dockerStage('Container Build') {
-    sh "docker build -t quay.io/powerhome/${application}:${tag} ."
-    sh "docker push quay.io/powerhome/${application}:${tag}"
+    sh "docker build -t ${appImage} ."
+    sh "docker push ${appImage}"
   }
 
   stage('Test') {
```
Comment (benlangfeld) [inline Jenkinsfile]:
```suggestion
  app.dockerStage('Test') {
```

File: Jenkinsfile (hunk 1 above)
Comment (garettarrowood) [reply Jenkinsfile]: I don't know, I'm just trying things TBH.

**Comment by garettarrowood** [comment]
Screen Shot 2019-10-01 at 12 26 51 PM

**Comment by garettarrowood** [comment]
Screen Shot 2019-10-01 at 12 29 11 PM

**Pull request merged**